### ISSUE TITLE
Allow adding USB2 controller

### ISSUE THREAD

**Issue by jschwartzenberg** [vagrant-libvirt/vagrant-libvirt#597]
### Steps to reproduce
1. Create a Vagrant box with Spice, QXL, etc.
2. Search for a way to add spicevmc and a USB2 controller
3. Can't find anything about it in the readme
### Expected behaviour

Have some options documented to add specific hardware such as spicevmc and usb2 hardware.
### Actual behaviour

Can't find it in the documentation, I suppose it's not implemented yet.
### System configuration

**OS/Distro version:**:
CentOS 7 host, Fedora 23 guest

**Libvirt version:**
1.2.17

**Output of `vagrant version; vagrant plugin list`:**
Installed Version: 1.8.1
Latest Version: 1.8.1

You're running an up-to-date version of Vagrant!
vagrant-libvirt (0.0.33)
vagrant-share (1.1.5, system)

**A Vagrantfile to reproduce the issue:**

```
VAGRANTFILE_API_VERSION = "2"
Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
config.vm.box = "fedora/23-cloud-base"
  config.vm.provider "libvirt" do |libvirt|
    libvirt.memory = 2048
    libvirt.graphics_type = "spice"
    libvirt.video_type = "qxl"
    libvirt.cpu_feature :name => 'avx', :policy => 'disable'
  end
end
```

**Are you using upstream vagrant package or your distros package?**
Upstream

**Comment by infernix** [comment]
This is what libvirt adds by default:
```xml
    <controller type='usb' index='0'>
      <address type='pci' domain='0x0000' bus='0x00' slot='0x01' function='0x2'/>
    </controller>
```

When I use virt-manager to switch that to USB2, libvirt changes that into:
```xml
    <controller type='usb' index='0' model='ich9-ehci1'>
      <address type='pci' domain='0x0000' bus='0x00' slot='0x08' function='0x7'/>
    </controller>
    <controller type='usb' index='0' model='ich9-uhci1'>
      <master startport='0'/>
      <address type='pci' domain='0x0000' bus='0x00' slot='0x08' function='0x0' multifunction='on'/>
    </controller>
    <controller type='usb' index='0' model='ich9-uhci2'>
      <master startport='2'/>
      <address type='pci' domain='0x0000' bus='0x00' slot='0x08' function='0x1'/>
    </controller>
    <controller type='usb' index='0' model='ich9-uhci3'>
      <master startport='4'/>
      <address type='pci' domain='0x0000' bus='0x00' slot='0x08' function='0x2'/>
    </controller>
```

For spicevmc, I assume you mean this:
```xml
    <controller type='virtio-serial' index='0'>
      <address type='pci' domain='0x0000' bus='0x00' slot='0x07' function='0x0'/>
    </controller>
<!--- ... --->
    <channel type='spicevmc'>
      <target type='virtio' name='com.redhat.spice.0'/>
      <address type='virtio-serial' controller='0' bus='0' port='1'/>
    </channel>
```

Can you confirm that this is what you are looking for?

**Comment by jschwartzenberg** [comment]
Yes, it seems that that's what's needed.

**Comment by infernix** [comment]
It's not currently supported, so I'll keep flagged as enhancement. If you could elaborate on the use case a bit that would help with documentation.

**Comment by jschwartzenberg** [comment]
These two components are necessary to provide full SPICE functionality.

**Comment by infernix** [comment]
Will keep this open for USB2 support; channel support is now merged.

**Comment by rgl** [comment]
I would also like to have USB 2.0 support. I need to pass a webcam into the VM. Without USB 2.0 support I get this error:

```
==> default: Starting domain.
There was an error talking to Libvirt. The error message is shown
below:

Call to virDomainCreateWithFlags failed: internal error: qemu unexpectedly closed the monitor: 2017-10-01T12:19:17.063773Z qemu-system-x86_64: -device usb-host,hostbus=2,hostaddr=3,id=hostdev0,bus=usb.0,port=2: Warning: speed mismatch trying to attach usb device "host:2.3" (high speed) to bus "usb.0", port "2" (full speed)
2017-10-01T12:19:18.824262Z qemu-system-x86_64: -device usb-host,hostbus=2,hostaddr=3,id=hostdev0,bus=usb.0,port=2: failed to open host usb device 2:3
```
